docs: wcag 3.3.2 full page (#2362)
nl-design-system/documentatie: docs/wcag/3.3.02.mdx

hide_table_of_contents: false
sidebar_label: 3.3.2 Labels of instructies
pagination_label: WCAG-succescriterium 3.3.2 Labels of instructies

> description: Geef invoervelden een label of uitleg.

Reviewer: ```suggestion
description: Geef invoervelden een label  en indien nodig een uitleg.
```

Author: Ik vond deze ook flauw, maar het criterium is "of". Ik wil niet te veel advocaat spelen met deze teksten, maar de betekenis veranderd wel heel erg met je voorstel ben ik bang.

  {"WCAG-succescriterium 3.3.2 Labels of instructies"}
</WcagHeadingGroup>

## In het kort

<dl>
  <div class="dl__item">
    <dt class="dl__term">Doel</dt>
    <dd class="dl__definition">Gebruikers weten wat ze in moeten voeren.</dd>
  </div>
  <div class="dl__item">
    <dt class="dl__term">Wat te doen</dt>
    <dd class="dl__definition">Geef een label of uitleg bij invoer.</dd>
  </div>
  <div class="dl__item">
    <dt class="dl__term">Waarom het belangrijk is</dt>
    <dd class="dl__definition">Iedereen, vooral mensen met cognitieve beperkingen, weet waar men aan toe is.</dd>
  </div>
</dl>

## Uitleg

<Summary />

Als er ergens invoer nodig is, is het voor iedereen prettig om te weten wat er verwacht wordt. Labels en uitleg kunnen hierbij helpen. Labels zijn vaak ook een manier om een naam te geven aan een invoerveld. Denk hierbij aan termen als:

- Voornaam
- Plaatsnaam

> - Wachtwoord

Reviewer: ```suggestion
- Wachtwoord, of een hangslot-icoon
- Telefoonnummer, of een telefoon-icoon
- Een vergrootglas-icoon bij een zoekfunctie

```

Instructies vertellen meer over wat er verwacht wordt van de invoer, zoals de manier van spellen of schrijven. Zo kunnen er voor de invoer van een nieuw wachtwoord eisen zijn, of is er voor een datum een bepaalde manier van invoeren.

## Gerelateerde NL Design System-richtlijnen

Dit succescriterium past bij bijna alle richtlijnen rond formulieren.

- [Formulieren](https://nldesignsystem.nl/richtlijnen/formulieren)
- Formulieren - [Labels](https://nldesignsystem.nl/richtlijnen/formulieren/labels/)

## Hoe te testen

- Zoek de invoervelden en formulieren.
- Controleer van elk veld of er een beschrijvend label of behulpzame uitleg bij is.

##Tips

- Het behalen van dit criterium kan onder andere ook helpen bij [2.4.6 Koppen en labels](/wcag/2.4.6) en [4.1.2 Naam, rol, waarde](/wcag/4.1.2).
- Zorg voor zowel labels als uitleg om fouten te voorkomen.

## Veelgemaakte fouten

### Fout: Labels en/of uitleg zijn niet verbonden aan de invoer.

> Voor [1.3.1 Info en relaties](/wcag/1.3.1) is het belangrijk dat labels en uitleg niet alleen aanwezig zijn, maar ook in de code aan elkaar verbonden zijn. De relatie die je ziet moet ook in de code aanwezig zijn. Dit kan met het `<label>`-element, maar ook met `aria-labelledby` en `aria-describedby`. Foutmelding moeten ook gekoppeld zijn aan het invoerveld. Dit kan met `aria-describedby`.

Reviewer: ```suggestion
Voor [1.3.1 Info en relaties](/wcag/1.3.1) is het belangrijk dat labels en uitleg niet alleen aanwezig zijn, maar ook in de code aan elkaar verbonden zijn. De relatie die je ziet moet ook in de code aanwezig zijn. Dit kan met het `<label>`-element, maar ook met `aria-labelledby`. Foutmeldingen kun je ook koppelen het invoerveld. Dit kan met `aria-describedby`.
```

Author: Moet dit niet vrijwel altijd? Zo vaak dat we het hier kunnen noemen, en voor uitzonderingen kunnen verwijzen naar de originele tekst?

- een nieuw wachtwoord dat lang genoeg moet zijn
- wel of geen spatie tussen de letters en cijfers van een postcode
- de manier waarop een datum geschreven moet worden

Als de invoer aan bepaalde eisen moet voldoen, is het een goed idee om een uitleg toe te voegen.

## Gerelateerde NL Design System-richtlijnen

Dit succescriterium past bij bijna alle richtlijnen rond formulieren.

- [Formulieren](https://nldesignsystem.nl/richtlijnen/formulieren)
- Formulieren - [Labels](https://nldesignsystem.nl/richtlijnen/formulieren/labels/)

## Hoe te testen

- Zoek de invoervelden en formulieren.
- Controleer van elk veld of er een beschrijvend label of behulpzame uitleg bij is.

##Tips

- Het behalen van dit criterium kan onder andere ook helpen bij [2.4.6 Koppen en labels](/wcag/2.4.6) en [4.1.2 Naam, rol, waarde](/wcag/4.1.2).
- Zorg voor zowel labels als uitleg om fouten te voorkomen.
- Als een nieuw wachtwoord aan bepaalde eisen moet voldoen, maak die dan meteen zichtbaar.

## Veelgemaakte fouten

### Fout: Labels en/of uitleg zijn niet verbonden aan de invoer.

> Voor [1.3.1 Info en relaties](/wcag/1.3.1) is het belangrijk dat labels en uitleg niet alleen aanwezig zijn, maar ook in de code aan elkaar verbonden zijn. De relatie die je ziet moet ook in de code aanwezig zijn. Dit kan met het `<label>`-element, maar ook met `aria-labelledby`. Foutmeldingen kun je ook koppelen het invoerveld. Dit kan met `aria-describedby`.

Reviewer: ```suggestion
Voor [1.3.1 Info en relaties](/wcag/1.3.1) is het belangrijk dat labels en uitleg niet alleen aanwezig zijn, maar ook in de code aan elkaar verbonden zijn. De relatie die je ziet moet ook in de code aanwezig zijn. Dit kan met het `<label>`-element, maar ook met `aria-labelledby`. Foutmeldingen kun je ook koppelen aan het invoerveld. Dit kan met `aria-describedby`.
```
